app/models/case/sar/offender_complaint.rb

```diff
@@ -0,0 +1,306 @@
+                 internal_deadline: :date,
+                 other_subject_ids: :string,
+                 previous_case_numbers: :string,
+                 prison_number: :string,
+                 recipient: :string,
+                 subject_address: :string,
+                 request_dated: :date,
+                 requester_reference: :string,
+                 subject_aliases: :string,
+                 subject_full_name: :string,
+                 subject_type: :string,
+                 third_party_relationship: :string,
+                 third_party: :boolean,
+                 third_party_company_name: :string,
+                 late_team_id: :integer,
+                 third_party_name: :string,
+                 number_final_pages: :integer,
+                 number_exempt_pages: :integer
+
+  attribute :number_final_pages, :integer, default: 0
+  attribute :number_exempt_pages, :integer, default: 0
+
+  enum subject_type: {
+    offender: 'offender',
+    ex_offender: 'ex_offender',
+    detainee: 'detainee',
+    ex_detainee: 'ex_detainee',
+    probation_service_user: 'probation_service_user',
+    ex_probation_service_user: 'ex_probation_service_user',
+  }
+
+  enum recipient: {
+    subject_recipient:  'subject_recipient',
+    requester_recipient: 'requester_recipient',
+    third_party_recipient: 'third_party_recipient',
+  }
+
+  has_paper_trail only: [
+    :name,
+    :email,
+    :postal_address,
+    :properties,
+    :received_date,
+  ]
+
+  has_many :data_requests, dependent: :destroy, foreign_key: :case_id, as: :offender_case
+  accepts_nested_attributes_for :data_requests
+
+  validates :third_party,          inclusion: { in: [true, false], message: "can't be blank" }
+  validates :flag_as_high_profile, inclusion: { in: [true, false], message: "can't be blank" }
+  validates :date_of_birth, presence: true
+
+  validates_presence_of :subject_address
+
+  validates :subject_full_name, presence: true
+  validates :subject_type, presence: true
+  validates :recipient, presence: true
+  validate :validate_date_of_birth
+  validate :validate_received_date
+  validate :validate_third_party_names
+  validate :validate_recipient
+  validate :validate_third_party_relationship
+  validate :validate_third_party_address
+  validate :validate_request_dated
+  validates :number_final_pages,
+            numericality: { only_integer: true, greater_than: -1,
+                            message: 'must be a positive whole number' }
+  validates :number_exempt_pages,
+            numericality: { only_integer: true, greater_than: -1,
+                            message: 'must be a positive whole number' }
+  validate :validate_third_party_states_consistent
+
+  before_validation :ensure_third_party_states_consistent
+  before_validation :reassign_gov_uk_dates
+  before_save :set_subject
+  before_save :use_subject_as_requester,
+              if: -> { name.blank? }
+
+  def validate_third_party_states_consistent
```

Comment:
I think a pragmatic thing we could do in this new type is to extract these validations out into `app/validators` using `ActiveModel::EachValidator` as a base class. Might be able to DRY things out a little with OffenderSars as well with this.